projects/graph/src/draw.py

```diff
@@ -2,13 +2,172 @@
 General drawing methods for graphs using Bokeh.
 """
 
+# necessary imports
+
+from random import choice, random
 from bokeh.io import show, output_file
 from bokeh.plotting import figure
-from bokeh.models import (GraphRenderer, StaticLayoutProvider, Circle, LabelSet,
-                          ColumnDataSource)
+from bokeh.models import (
+    GraphRenderer,
+    StaticLayoutProvider,
+    Circle,
+    LabelSet,
+    ColumnDataSource,
+    VeeHead,
+    Arrow,
+)
 
 
 class BokehGraph:
     """Class that takes a graph and exposes drawing methods."""
-    def __init__(self):
-        pass  # TODO
+    # initial Bokeh graph parameters
+    def __init__(
+        self,
+        graph,
+        title="Sample Graph for Lambda Project",
+        # axis width and height
+        width=20,
+        height=20,
+        # options to display axis and/or grid
+        show_axis=True,
+        show_grid=False,
+        # vertex/node size
+        circle_size=25,
+    ):
+    # check if graph has vertices
+        if not graph.vertices:
+            raise Exception("Graph should contain vertices!")
+        self.graph = graph
+
+        # Setup plot
+        self.width = width
+        self.height = height
+        self.pos = {}  # dict to map vertices by x, y positions
+        # new plot with title and ranges
+        """The bokeh.plotting interface provides a Figure class to help with assembling all the necessary objects.
+            the figure() function creates Figure objects."""
+        self.plot = figure(
+            title=title,
+            # Ranges describe the data-space bounds of a plot, in this case, from 0 to the graphs width and height
```

Comment:
Remember Python doesn't like long lines or bad whitespace. In fact the official PEP8 documentation says that lines should be no longer that 88 characters. I would consider refactoring to be more Pythonic, it will definitely help you dive deeper into the language and improve your skills. It's always better to see what best practices are for each language.